// brainwave-frontend/src/pages/dashboard.jsx
import { HomeIcon, BookOpenIcon, CalendarIcon, Cog6ToothIcon, TrophyIcon, UsersIcon, CpuChipIcon, AcademicCapIcon, CheckBadgeIcon, BellIcon } from "@heroicons/react/24/outline";


// Sidebar items
const nav = [
  { icon: <HomeIcon className="h-6 w-6"/>, text: "Dashboard" },
  { icon: <BookOpenIcon className="h-6 w-6"/>, text: "My Courses" },
  { icon: <CpuChipIcon className="h-6 w-6"/>, text: "AI Assistant" },
  { icon: <AcademicCapIcon className="h-6 w-6"/>, text: "Library" },
  { icon: <CalendarIcon className="h-6 w-6"/>, text: "Calendar" },
  { icon: <TrophyIcon className="h-6 w-6"/>, text: "Leaderboard" },
  { icon: <UsersIcon className="h-6 w-6"/>, text: "Alumni Connect" },
  { icon: <Cog6ToothIcon className="h-6 w-6"/>, text: "Settings" },
];

// Example Data
const user = {
  name: "Alex Johnson",
  avatar: "/avatar.png",
  level: 15,
  xp: 1250,
  nextLevel: 1000,
  cgpa: 8.5,
  attendance: 92,
  pending: 3,
  libraryDue: 2,
  studentId: "ST2023001",
  sem: "6th Semester",
  rank: 5,
};
const courseProgress = [
  { name: "Web Development", grade: "A", progress: 75 },
  { name: "Data Structures", grade: "A+", progress: 80 },
  { name: "Database Management", grade: "B+", progress: 60 },
  { name: "Software Engineering", grade: "A+", progress: 90 },
];
const events = [
  { date: "25 Sep", title: "Database Midterm", type: "Exam" },
  { date: "28 Sep", title: "Web Dev Project Due", type: "Assignment Due" },
  { date: "2 Oct", title: "DS Assignment 3", type: "Assignment Due" },
];
const recent = [
  { desc: "Submitted Web Dev Assignment 2", time: "Sep 18, 2025" },
  { desc: "Completed Database Quiz", time: "Sep 17, 2025" },
  { desc: "Borrowed Clean Code from library", time: "Sep 15, 2025" },
];

export default function Dashboard() {
  return (
    <div className="min-h-screen bg-gradient-to-br from-[#edf7fa] to-[#e0f2fe] flex">
      {/* Sidebar */}
      <aside className="bg-white shadow-lg w-64 p-6 flex flex-col gap-6 sticky top-0 h-screen">
        <div className="font-black tracking-tight text-cyan-600 text-xl mb-6 flex items-center gap-2">
          <CpuChipIcon className="h-7 w-7 text-cyan-400" /> BrainWave
        </div>
        <nav className="flex flex-col gap-2">
          {nav.map((item, idx) => (
            <a key={idx} className={`flex items-center gap-3 px-3 py-2 rounded-lg transition hover:bg-cyan-100 font-medium ${idx===0?"bg-cyan-50 text-cyan-900": "text-gray-700"}`}>
              <span>{item.icon}</span>{item.text}
            </a>
          ))}
        </nav>
        <div className="mt-auto">
          <div className="flex items-center gap-3">
            <img src={user.avatar} className="w-10 h-10 bg-white rounded-full ring-2 ring-cyan-400" alt="avatar"/>
            <div>
              <div className="font-bold text-cyan-700">{user.name}</div>
              <div className="text-xs text-gray-400">Level {user.level}</div>
            </div>
          </div>
        </div>
      </aside>
      
      {/* Main Panel */}
      <main className="flex-1 px-10 py-8">
        {/* Topbar */}
        <div className="flex justify-between items-center mb-6">
          <div>
            <div className="text-lg text-gray-600">Welcome back,</div>
            <div className="text-2xl font-bold text-cyan-700">{user.name}!</div>
            <div className="text-sm text-gray-400">{user.sem} &bull; Student ID: <span className="font-bold">{user.studentId}</span></div>
          </div>
          <div className="flex gap-4 items-center">
            <input placeholder="Search courses, assignments, notes..." className="px-4 py-2 rounded-lg border text-sm bg-white"/>
            <button className="relative">
              <BellIcon className="h-6 w-6 text-gray-600"/>
              {/* Red dot for notifications */}
            </button>
            <img src={user.avatar} alt="avatar" className="h-10 w-10 rounded-full bg-white ring-2 ring-cyan-400 ml-4"/>
          </div>
        </div>
        
        {/* Stat Cards Row */}
        <div className="grid grid-cols-1 md:grid-cols-4 gap-6 mb-8">
          <div className="bg-white rounded-xl shadow flex flex-col items-center px-5 py-5">
            <CheckBadgeIcon className="h-6 w-6 text-cyan-400 mb-2" />
            <div className="text-lg font-bold text-cyan-700">{user.cgpa}</div>
            <div className="text-xs text-gray-400 mt-1">Current CGPA</div>
          </div>
          <div className="bg-white rounded-xl shadow flex flex-col items-center px-5 py-5">
            <CheckBadgeIcon className="h-6 w-6 text-emerald-400 mb-2" />
            <div className="text-lg font-bold text-emerald-600">{user.attendance}%</div>
            <div className="text-xs text-gray-400 mt-1">Attendance</div>
          </div>
          <div className="bg-white rounded-xl shadow flex flex-col items-center px-5 py-5">
            <CheckBadgeIcon className="h-6 w-6 text-yellow-400 mb-2" />
            <div className="text-lg font-bold text-yellow-700">{user.pending}</div>
            <div className="text-xs text-gray-400 mt-1">Pending Assignments</div>
          </div>
          <div className="bg-white rounded-xl shadow flex flex-col items-center px-5 py-5">
            <BookOpenIcon className="h-6 w-6 text-pink-400 mb-2" />
            <div className="text-lg font-bold text-pink-600">{user.libraryDue}</div>
            <div className="text-xs text-gray-400 mt-1">Library Books Due</div>
          </div>
        </div>

        {/* Academic Progress */}
        <div className="grid md:grid-cols-2 gap-10 mb-12">
          {/* Course Progress */}
          <div>
            <div className="font-bold text-lg text-cyan-800 mb-4">Course Progress</div>
            <div className="flex flex-col gap-5">
              {courseProgress.map((c, i) => (
                <div key={i} className="bg-white px-5 py-3 rounded-lg shadow flex justify-between items-center">
                  <div>
                    <div className="font-semibold text-cyan-700">{c.name}</div>
                    <div className="text-xs text-cyan-500">Grade: <span className="font-bold text-cyan-700">{c.grade}</span></div>
                  </div>
                  <div className="flex flex-col items-end">
                    <div className="text-xs text-gray-400 mb-1">{c.progress}% complete</div>
                    <div className="w-36 h-3 bg-gray-200 rounded-lg overflow-hidden">
                      <div
                        className={`h-3 rounded-lg ${c.progress>80?"bg-emerald-400":c.progress>60?"bg-yellow-400":"bg-cyan-200"}`}
                        style={{ width: `${c.progress}%` }}
                      ></div>
                    </div>
                  </div>
                </div>
              ))}
            </div>
          </div>

          {/* Upcoming Events */}
          <div>
            <div className="font-bold text-lg text-cyan-800 mb-4">Upcoming Events</div>
            <div className="flex flex-col gap-5">
              {events.map((e, i) => (
                <div key={i} className="bg-white px-5 py-3 rounded-lg shadow flex items-center gap-5">
                  <div className="bg-cyan-100 rounded-lg p-3 flex flex-col items-center justify-center">
                    <span className="text-xl font-extrabold text-cyan-600">{e.date.split(" ")[0]}</span>
                    <span className="text-xs text-cyan-500">{e.date.split(" ")[1]}</span>
                  </div>
                  <div>
                    <div className="font-bold text-cyan-800">{e.title}</div>
                    <div className="text-xs text-gray-400">{e.type}</div>
                  </div>
                </div>
              ))}
            </div>
          </div>
        </div>
        
        {/* Your Progress */}
        <div className="grid md:grid-cols-2 gap-8 mb-12">
          <div className="bg-white shadow rounded-xl px-8 py-6 flex flex-col items-center">
            <div className="mb-3">
              <span className="text-xs text-gray-400">Level</span>
              <span className="text-xl font-bold text-cyan-700 ml-2">{user.level}</span>
            </div>
            <div className="text-cyan-800 text-2xl font-extrabold mb-1">{user.xp} XP</div>
            <div className="w-full bg-gray-100 h-4 rounded-lg mt-2 mb-1 overflow-hidden">
              <div className="h-4 bg-cyan-400" style={{ width: `${(user.xp%user.nextLevel)/user.nextLevel*100 || 75}%` }}></div>
            </div>
            <div className="text-xs text-gray-400">{user.xp % user.nextLevel}/{user.nextLevel} to next level</div>
            <div className="mt-3 text-lg text-gray-600">🏆 <b>#{user.rank}</b> in class</div>
          </div>
          {/* Recent Activity */}
          <div className="bg-white shadow rounded-xl px-8 py-6">
            <div className="font-bold text-lg text-cyan-700 mb-4">Recent Activity</div>
            <ul className="space-y-2">
              {recent.map((r, i) => (
                <li key={i} className="flex justify-between text-cyan-700">
                  <span>{r.desc}</span>
                  <span className="text-xs text-gray-400">{r.time}</span>
                </li>
              ))}
            </ul>
          </div>
        </div>

      </main>
    </div>
  );
}
