// brainwave-frontend/src/pages/Profile.jsx
import React from "react";

const student = {
  name: "Student Name",
  email: "student@example.com",
  avatar: "/avatar.png", // Use your real avatar image path!
  phone: "+91 9876543210",
  course: "B.Tech Computer Science",
  year: "2025",
  college: "Brainwave Institute",
  bio: "Curious learner, passionate about technology, loves building stuff and collaborating with peers.",
};

const Profile = () => (
  <div className="min-h-screen flex items-center justify-center bg-brainwave-bg">
    <div className="bg-brainwave-primary-glass shadow-brainwave p-10 rounded-2xl w-full max-w-lg flex flex-col items-center">
      {/* Avatar section */}
      <img
        src={student.avatar}
        alt="avatar"
        className="w-28 h-28 rounded-full border-4 border-brainwave-secondary mb-6 bg-white object-cover"
      />

      {/* Name & Role */}
      <h2 className="text-2xl font-extrabold text-brainwave-secondary text-center mb-2">
        {student.name}
      </h2>
      <p className="text-brainwave-secondary text-center mb-4 opacity-80">{student.course} • {student.year}</p>
      <p className="text-brainwave-primary text-center mb-7">{student.college}</p>

      {/* Basic Info */}
      <div className="w-full flex flex-col gap-2 text-brainwave-secondary">
        <div className="flex justify-between">
          <span>Email:</span>
          <span className="text-brainwave-primary">{student.email}</span>
        </div>
        <div className="flex justify-between">
          <span>Phone:</span>
          <span className="text-brainwave-primary">{student.phone}</span>
        </div>
      </div>

      <hr className="my-6 w-full border-brainwave-secondary opacity-30" />

      {/* Bio */}
      <p className="text-center text-brainwave-secondary">{student.bio}</p>
    </div>
  </div>
);

export default Profile;
